satpy/tests/test_yaml_reader.py

```diff
@@ -1527,3 +1527,278 @@ def test_get_empty_segment_with_height(self):
+            "file_patterns": [
+                "{platform}-c-{start_time:%Y%m%d%H%M%S}-{end_time:%Y%m%d%H%M%S}-{segment:>02d}.nc",
+                "{platform}-d-{start_time:%Y%m%d%H%M%S}-{end_time:%Y%m%d%H%M%S}-{segment:>02d}.nc"]}
+
+    gsyr = GEOSegmentYAMLReader(
+            {"reader": {
+                "name": "island-reader"},
+             "file_types": {
+                 "m9g": fake_filetype_info,
+                 "mag": ft_info_2}}, preload=True)
+
+
+    # filename info belonging to fake_simple_nc_file
+
+    fn_info = {"platform": "M9a", "start_time": datetime(2100, 1, 1, 5, 30),
+               "end_time": datetime(2100, 1, 1, 5, 31), "segment": 1}
+
+    with unittest.mock.patch("appdirs.user_cache_dir") as au:
+        au.return_value = os.fspath(tmp_path / "cache")
+        # prepare cache files
+        dph = dummy_preloadable_handler(os.fspath(fake_simple_nc_file),
+                                        fn_info, fake_filetype_info)
+        for i in range(2, 4):  # disk cache except for nr. 1
+            fn = (tmp_path / "cache" / "satpy" / "preloadable" /
+                  "DummyPreloadableHandler" /
+                  f"M9G-a-99991231235959-99991231235959-{i:>02d}.pkl")
+            fn.parent.mkdir(exist_ok=True, parents=True)
+            dph.store_cache(os.fspath(fn))
+
+        fhs = gsyr.create_filehandlers([os.fspath(fake_simple_nc_file)])
+        assert len(fhs["m9g"]) == 3
+
+
+def test_preloaded_instances_requirement(
+        tmp_path, fake_gsyreader, fake_simple_nc_file,
+        dummy_preloadable_handler, fake_filetype_info):
+    """Test that pre-loading instances fails if there is a required tag."""
+    from satpy.readers.yaml_reader import GEOSegmentYAMLReader
+
+    ft_info = {**fake_filetype_info,
+               "requires": ["pergola"]}
+
+    gsyr = GEOSegmentYAMLReader(
+            {"reader": {
+                "name": "alicudi"},
+             "file_types": {
+                 "m9g": ft_info}}, preload=True)
+    g = gsyr._new_filehandler_instances(
+            ft_info,
+            [(os.fspath(fake_simple_nc_file),
+              {"platform": "M9G",
+               "start_time": datetime(2100, 1, 1, 5, 30, ),
+               "end_time": datetime(2100, 1, 1, 5, 31, ),
+               "segment": 1})])
+    with pytest.raises(ValueError, match="Unable to preload"):
+        list(g)
+
+
+def test_preloaded_instances_not_implemented(tmp_path, fake_gsyreader,
+                                             fake_filetype_info):
+    """Test that pre-loading instances fails if it is not implemented."""
+    ft_info = {**fake_filetype_info, "requires": ["pergola"]}
+
+    # Second argument irrelevant, as this part of the method should not be
+    # reached
+    g = fake_gsyreader._new_preloaded_filehandler_instances(ft_info, None)
+    with pytest.raises(NotImplementedError, match="Pre-loading not implemented"):
+        list(g)
+
+
+def test_get_cache_filename(tmp_path):
```

Comment:
Could this test be split in multiple smaller ones?

Reply:
Split in three in 30e9755ff